app/src/main/java/com/example/pokemonapp/rv/PokemonListAdapter.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.pokemonapp.rv
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.AsyncListDiffer
+import androidx.recyclerview.widget.RecyclerView
+import com.example.pokemonapp.databinding.ItemPokemonBinding
+import com.example.pokemonapp.model.Pokemon
+
+class PokemonListAdapter : RecyclerView.Adapter<PokemonViewHolder>() {
+
+
+       private var items = emptyList<Pokemon>()
+       private val differ = AsyncListDiffer(this, DiffCallback)
+       var onClick: (Pokemon) -> Unit = { }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PokemonViewHolder {
+        val view = ItemPokemonBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return PokemonViewHolder(view)
+    }
+    override fun getItemCount(): Int = differ.currentList.size
```

Comment:
```suggestion
    override fun getItemCount(): Int = items.size
```
ну или зачем вообще тогда сохранять список элементов? Тогда ведь можно всегда из диффера забирать)